packages/core/src/xkeys.ts

```diff
@@ -112,6 +112,23 @@ export class XKeys extends EventEmitter {
 				for (let i = 0; i < 15; i++) {
 					data[i] = rdData[i]
 				}
+			} else if (deviceInfo.productId === 255) {
+				// Note: The VEC Footpedal is an older device, which doesn't follow the rest of xkeys data structure.
+				// To make it easy for us, we'll just remap the data to work for us.
+
+				const rdData = new Uint8Array(32)
+				rdData[0] = 0 // this sets the Unit ID to 0 always
+				if (!this._firmwareVersionIsSet) {
+					rdData[1] = 214 // Fake initial message to set _firmwareVersion
+				} else if (!this._unitIdIsSet) {
+					rdData[1] = 3 // Fake initial message to set _unitId
```

Comment:
The first up and down events seem to get swallowed, can we not force extra messages here which I assume would solve this? Presumably this is also the same for the RailDriver?

Reply:
Am I correct in assuming that this device doesn't really have the concept of a unitId? (That would make sense, since it won't accept the inititial write of `_getVersion()` and so won't return a unitId.)

If so, I suggest that we solve this by adding

```
if (this.deviceInfo.productId == 255) {
  this._unitId = 0 // VEC Footpanel doesn't support unitIds
  this._unitIdIsSet = true
}
```
into the `init()` method.

Reply:
Given it only ever returns the two bytes, and doesn't accept any writes, I don't think so! Although conversely I've never worked out what the second byte is for (some more generic underlying product/interface)

The device has no serial number via lsusb either (but maybe that's common to all xkeys?).

> If so, I suggest that we solve this by adding

Do you think this would solve it missing the first keypress and release too when run as `node packages/node/examples/basic-log-all-events.js`?